modules/tensor_mechanics/test/src/materials/SS316HLAROMANCEStressUpdateTest.C

```diff
@@ -7,69 +7,80 @@
 //* Licensed under LGPL 2.1, please see LICENSE for details
 //* https://www.gnu.org/licenses/lgpl-2.1.html
 
-//  saturation as a function of effective saturation, and its derivs wrt effective saturation
-//
-#include "SS316LAROMData.h"
+#include "SS316HLAROMANCEStressUpdateTest.h"
 
-registerMooseObject("TensorMechanicsApp", SS316LAROMData);
+registerADMooseObject("TensorMechanicsTestApp", SS316HLAROMANCEStressUpdateTest);
 
-template <>
-InputParameters
-validParams<SS316LAROMData>()
-{
-  InputParameters params = validParams<LAROMData>();
-  params.addClassDescription("UserObject that stores the relevant data for the reduced order model "
-                             "for SS316. Must be used in conjuction with ADROMCreepStressUpdate");
-  return params;
-}
-
-SS316LAROMData::SS316LAROMData(const InputParameters & parameters) : LAROMData(parameters) {}
-
-unsigned int
-SS316LAROMData::getStressIndex() const
-{
-  return 2;
-}
-
-unsigned int
-SS316LAROMData::getDegree() const
-{
-  return 4;
-}
+defineADValidParams(
+    SS316HLAROMANCEStressUpdateTest,
+    ADLAROMANCEStressUpdateBase,
+    params.addClassDescription("LAROMANCE creep update model for SS316H");
 
-Real
-SS316LAROMData::getMaxRelativeMobileInc() const
-{
-  return 0.5;
-}
-
-Real
-SS316LAROMData::getMaxRelativeImmobileInc() const
-{
-  return 0.5;
-}
+    // Override defaults for material specific parameters below
+    params.addRangeCheckedParam<unsigned int>(
+        "stress_index",
+        2,
+        "stress_index > 0",
+        "Index corresponding to the trial stress for in the input vector");
+    params.addRangeCheckedParam<Real>("initial_mobile_dislocation_density",
+                                      5.0e12,
+                                      "initial_mobile_dislocation_density >= 0.0",
```

Comment:
Since this is a test object,  not the base class, how about setting the range check to the lower range (the 5.0e12 here and the 1.0e11 for the immobile below)

Reply:
Yeah, I can really do this for the lower and upper ranges for the dislocations, good idea.